app/grandchallenge/components/views.py

```diff
@@ -84,3 +89,50 @@ def get_queryset(self):
 
     def get_result_label(self, result):
         return result.title
+
+
+class InterfaceProcessingMixin:
+    def process_data_for_object(self, data):
+        raise NotImplementedError
+
+    def form_valid(self, form):
+        form.instance = self.process_data_for_object(form.cleaned_data)
+        super().form_valid(form)
+        # since this is an HTMX view, we need to trigger
+        # the redirect through inclusion of the HX headers
+        # HttpResponseRedirect does not support the inclusion of these custom headers
+        messages.add_message(
+            self.request, messages.SUCCESS, self.success_message
+        )
+        return HttpResponse(
+            self.get_success_url(),
+            headers={
+                "HX-Redirect": self.get_success_url(),

```

Comment:
Instead of `self.get_success_url()` I think this needs to be `response.url` with `response = super().form_valid(form)`.